### PR TITLE
Trunk 5521 upgrade javax libs

### PR DESCRIPTION
**Pull request by Irenyak1**
## Description of what I changed

I updated the following libraries

   1.javax.mail:mail … 1.4.1 -> 1.4.7
   2.javax.servlet:javax.servlet-api … 3.0.1 -> 4.0.1
   3.javax.servlet:jsp-api … 2.0 -> 2.3.3
   4.javax.servlet:jstl … 1.1.2 -> 1.2
   5.javax.validation:validation-api … 1.0.0.GA -> 2.0.1.Final

## Issue I worked on
https://issues.openmrs.org/browse/TRUNK-5521

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -374,7 +375,7 @@
 			<dependency>
 				<groupId>javax.mail</groupId>
 				<artifactId>mail</artifactId>
-				<version>1.4.1</version>
```
Comment (dkayiwa) [inline pom.xml]: Is this part of the ticket?

File: pom.xml [hunk 2]
```diff
@@ -99,14 +99,15 @@
 				<scope>provided</scope>
 			</dependency>
 			<dependency>
-				<groupId>javax.servlet</groupId>
-				<artifactId>jsp-api</artifactId>
-				<version>2.0</version>
+			    <groupId>javax.servlet.jsp</groupId>
+			    <artifactId>javax.servlet.jsp-api</artifactId>
+			    <version>2.3.3</version>
+			    <scope>provided</scope>
```
Comment (dkayiwa) [inline pom.xml]: Is adding the scope as provided, part of the ticket?

Comment (Irenyak1) [reply pom.xml]: @dkayiwa  it is not part of the ticket but I thought they move together, as it appears in the central maven repo.  Do I have to remove  the scope?

File: pom.xml (hunk 1 above)
Comment (Irenyak1) [reply pom.xml]: yes it is @dkayiwa

File: pom.xml (hunk 2 above)
Comment (Irenyak1) [reply pom.xml]: Hello @dkayiwa  after removing scope, it worked like magic, the build is passing now. Thank you so much.

**Comment by Irenyak1** [comment]
Hello @dkayiwa  please may you look at this PR, I have edited out the scope and it is building.

**Comment by Irenyak1** [comment]
Hello @dkayiwa  and @mozzy11 please review my PR, I have upgraded all the javax libraries and are building successfully.

**Comment by mozzy11** [comment]
@Irenyak1  can you try to remove the unncesary white spaces ??

**Comment by Irenyak1** [comment]
@mozzy11  after removing the white space still my coverage has reduced, how do I go about it?

File: web/pom.xml [hunk 3]
```diff
@@ -50,9 +50,9 @@
          <artifactId>javax.servlet-api</artifactId>
       </dependency>
       <dependency>
-         <groupId>javax.servlet</groupId>
-         <artifactId>jsp-api</artifactId>
-      </dependency>
+	    <groupId>javax.servlet.jsp</groupId>
+	    <artifactId>javax.servlet.jsp-api</artifactId>
```
Comment (mozzy11) [inline web/pom.xml]: seeing some undesired white spaces

File: pom.xml [hunk 4]
```diff
@@ -95,18 +95,18 @@
...
-				<version>3.0.1</version>
+				<version>4.0.1</version>
 				<scope>provided</scope>
 			</dependency>
 			<dependency>
-				<groupId>javax.servlet</groupId>
-				<artifactId>jsp-api</artifactId>
-				<version>2.0</version>
+			    <groupId>javax.servlet.jsp</groupId>
+			    <artifactId>javax.servlet.jsp-api</artifactId>
```
Comment (mozzy11) [inline pom.xml]: undesired white spaces

File: webapp/pom.xml [hunk 5]
```diff
@@ -42,9 +42,9 @@
                <artifactId>servlet-api</artifactId>
             </exclusion>
             <exclusion>
-               <groupId>javax.servlet</groupId>
-               <artifactId>jsp-api</artifactId>
-            </exclusion>
+			    <groupId>javax.servlet.jsp</groupId>
+			    <artifactId>javax.servlet.jsp-api</artifactId>
```
Comment (mozzy11) [inline webapp/pom.xml]: undesired spaces

File: web/pom.xml (hunk 3 above)
Comment (Irenyak1) [reply web/pom.xml]: Please @mozzy11 may you point to the lines that have white space?

File: pom.xml (hunk 4 above)
Comment (Irenyak1) [reply pom.xml]: Please @mozzy11 may you point to the lines that have white space?

File: webapp/pom.xml (hunk 5 above)
Comment (Irenyak1) [reply webapp/pom.xml]: Please @mozzy11 may you point to the lines that have white space?

File: web/src/main/java/org/openmrs/web/filter/GZIPRequestStream.java [hunk 6]
```diff
@@ -43,5 +44,22 @@ public int read() throws IOException {
 	public int read(byte[] b) throws IOException {
 		return zipInput.read(b);
 	}
+
+	@Override
+	public boolean isFinished() {
```
Comment (dkayiwa) [inline web/src/main/java/org/openmrs/web/filter/GZIPRequestStream.java]: Are these methods part of the upgrade?

Comment (Irenyak1) [reply web/src/main/java/org/openmrs/web/filter/GZIPRequestStream.java]: yes @dkayiwa  they are part of the upgrqade

Comment (Irenyak1) [reply web/src/main/java/org/openmrs/web/filter/GZIPRequestStream.java]: @dkayiwa  were you able to look through this PR, is it ready for merging, or is there  some more work I need to do? Let me know.

File: pom.xml (hunk 1 above)
Comment (dkayiwa) [reply pom.xml]: I have just seen this on the ticket. But am wondering why should these libraries be upgrade together in the same pull request? Are they related?

Comment (Irenyak1) [reply pom.xml]: I was following the ticket strictly. However, I can still separate them and do each library with it's pull request. What do you advise @dkayiwa

Comment (Irenyak1) [reply pom.xml]: I think the issue was about Javax, so all those lbrary under javax were grouped together in one ticket. @dkayiwa . Though I upgraded one at time until they all compiled locally then I made a joint PR.

File: web/src/main/java/org/openmrs/web/filter/GZIPRequestStream.java (hunk 6 above)
Comment (Irenyak1) [reply web/src/main/java/org/openmrs/web/filter/GZIPRequestStream.java]: Hello @dkayiwa is there still something stopping this from being merged?

**Comment by dkayiwa** [comment]
The pull request changes do not match with the ticket.

**Comment by Irenyak1** [comment]
@dkayiwa  I have added a comment on the ticket. Please review my PR.

File: web/src/main/java/org/openmrs/web/filter/GZIPRequestStream.java [hunk 7]
```diff
@@ -43,5 +44,22 @@ public int read() throws IOException {
...
+		// TODO Auto-generated method stub
+		return false;
```
Comment (mozzy11) [inline web/src/main/java/org/openmrs/web/filter/GZIPRequestStream.java]: @Irenyak1  why return false here?

File: web/src/main/java/org/openmrs/web/filter/GZIPRequestStream.java [hunk 8]
```diff
@@ -43,5 +44,22 @@ public int read() throws IOException {
...
+	}
+
+	@Override
+	public boolean isReady() {
+		// TODO Auto-generated method stub
+		return false;
```
Comment (mozzy11) [inline web/src/main/java/org/openmrs/web/filter/GZIPRequestStream.java]: @Irenyak1  why return false here?

File: web/src/main/java/org/openmrs/web/filter/GZIPResponseStream.java [hunk 9]
```diff
@@ -163,4 +165,15 @@ public boolean closed() {
 	public void reset() {
 		//noop
 	}
+
+	@Override
+	public boolean isReady() {
+		// TODO Auto-generated method stub
+		return false;
```
Comment (mozzy11) [inline web/src/main/java/org/openmrs/web/filter/GZIPResponseStream.java]: @Irenyak1 same here , why return false ?

**Comment by dkayiwa** [comment]
@Irenyak1 are you still working on this?

**Comment by Irenyak1** [comment]
@dkayiwa  yes I am still working on it.

**Comment by gitcliff** [comment]
@Irenyak1 you can also squash your commits into one

**Comment by Irenyak1** [comment]
Hello @mozzy11  and @dkayiwa  please review my PR when you find some time.

**Comment by Irenyak1** [comment]
Hello @mozzy11  and @dkayiwa  please review my PR when you find some time.

**Comment by dkayiwa** [comment]
Closing because of https://github.com/openmrs/openmrs-core/commit/4af8d2d9969a49f2efad7d3b0a458450809314dd

**Pull request closed without merging**